### PR TITLE
feat(ci): introduce better caching of node modules using gitlabs cache

### PR DESCRIPTION
**Pull request by vdovhanych**
This should improve caching bavior of node modules and prevent overwriting those modules with different package versions.

### CODE REVIEW

File: ci/build.yml [hunk 1]
```diff
@@ -72,6 +76,10 @@ suite-web build stable codesign:
 ## Build jobs
 suite-web-landing build dev:
   stage: build
+  cache:
+    key: "$CI_JOB_NAME-$CI_COMMIT_REF_SLUG"
+    paths:
+      - node_modules/
```
Comment (matejkriz) [inline ci/build.yml:93]: I'm not sure, do we want to cache node_modules or `.yarn` (cache-folder)

File: ci/build.yml [hunk 2]
```diff
@@ -56,6 +56,10 @@ suite-web build stable codesign:
     - darwin
   variables:
     IS_CODESIGN_BUILD: "true"
+  cache:
+    key: "$CI_JOB_NAME-$CI_COMMIT_REF_SLUG"
```
Comment (matejkriz) [inline ci/build.yml]: do we need to cache per job? why not cache per branch only?

**Review by matejkriz — COMMENTED** [review]
Do we have a single runner for all our jobs? Or some kind of distributed/shared caching?

File: ci/build.yml (hunk 1 above)
Comment (vdovhanych) [reply ci/build.yml:93]: I assume the yarn install cache option manages `.yarn`, and I don't recall that we had any issues with the `.yarn`, but it's not a problem to add it here and cache it in the same way.

File: ci/build.yml (hunk 2 above)
Comment (vdovhanych) [reply ci/build.yml]: Yeah, we can dump the $CI_JOB_NAME key and leave it for the whole branch. Good point

**Comment by tsusanka** [comment]
Just ccing @mroz22 here as caching NPM stuff was somewhat painful some time ago.

File: ci/build.yml (hunk 2 above)
Comment (vdovhanych) [reply ci/build.yml]: i have removed the per job cache and left only branch one

File: ci/test.yml [hunk 3]
```diff
@@ -16,6 +16,10 @@
     ## when debugging or developing tests it does not make sense to have retries,
     ## in other cases retries are useful to avoid occasional failures due to flaky tests
     ALLOW_RETRY: 1
+  cache:
+    key: "$CI_COMMIT_REF_SLUG"
+    paths:
+      - node_modules/
```
Comment (matejkriz) [inline ci/test.yml:22]: missing `.yarn` path

**Review by matejkriz — APPROVED** [review]
🎉

**Pull request merged**